apps/synapse-portal-framework/src/utils/useSetCanonicalUrl.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react'
+
+/**
+ * Hook to set the canonical URL of the page.
+ *
+ * The canonical URL helps search crawlers understand which URL(s) to index when there are multiple URLs that point to
+ * the same content.
+ * @param canonicalUrl
+ */
+export function useSetCanonicalUrl(canonicalUrl?: string) {
```

Comment:
One hypothesis is that Google is not indexing pages in our portals because they do not identify them as unique. Adding the canonical URL to the document head could help Google dedifferentiate duplicate pages, and also identify unique pages.